vulture/core.py

```diff
@@ -301,8 +360,14 @@ def by_size(item):
             + self.unreachable_code
         )
 
+        results_without_noqa = [
+            obj for obj in unused_code if not _has_noqa(obj)
+        ]
```

Comment:
No need to create a filtered list. Simply filter objects below.

Reply:
Yeah, that improves readability. thanks!